File: 프로그래머스/unrated/181932. 코드 처리하기/코드 처리하기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string code)
{
    string answer = "";

    int mode = 0;
    for (int i = 0; i < code.size(); i++)
    {
        if (mode == 0)
        {
            if (code[i] != '1')
            {
                if (!(i % 2))
                    answer += code[i];
            }
            else
                mode = 1;
        }
        else if (mode == 1)
        {
            if (code[i] != '1')
            {
                if (i % 2)
                    answer += code[i];
            }
            else
                mode = 0;
        }
    }
    if (!answer.size())
        answer = "EMPTY";

    return answer;
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {

        StringBuilder sb = new StringBuilder();
        
        int mode = 0;
        int idx = 0;
        for (char c : code.toCharArray()) {
            if (c == '1') {
                mode = mode == 0 ? 1 : 0;
            } else {
                if (mode == 0) {
                    if (idx % 2 == 0) {
                        sb.append(c);
                    }
                } else {
                    if (idx % 2 == 1) {
                        sb.append(c);
                    }
                }
            }
            idx++;
        }
        
        if (sb.length() == 0) return "EMPTY";
        
        return sb.toString();
    }
}